conans/client/runner.py

```diff
@@ -75,7 +75,7 @@ def _pipe_os_call(self, command, stream_output, log_handler, cwd):
             # piping both stdout, stderr and then later only reading one will hang the process
             # if the other fills the pip. So piping stdout, and redirecting stderr to stdout,
             # so both are merged and use just a single get_stream_lines() call
-            proc = Popen(command, shell=True, stdout=PIPE, stderr=STDOUT, cwd=cwd)
+            proc = Popen(command, shell=isinstance(command, six.string_types), stdout=PIPE, stderr=STDOUT, cwd=cwd)
```

Comment:
It is always true that for command args sequences, shell should be always False? From the docs:

> This can be useful if you are using Python primarily for the enhanced control flow it offers over most system shells and still want convenient access to other shell features such as shell pipes, filename wildcards, environment variable expansion, and expansion of ~ to a user’s home directory. 

Is it possible that users want to take advantage of shell features and provide sequences of args instead of strings?

Reply:
One reason I implemented it this way is to ensure that existing Conan scripts which pass strings with shell commands continue working just like that.

It does mean that when passing a sequence, shell features are unavailable. However, I don't think that's a significant disadvantage because the combination of passing a sequence *and* `shell=True` is implemented as [follows](https://docs.python.org/3.9/library/subprocess.html#popen-constructor):

> On POSIX with shell=True, […] `Popen` does the equivalent of:
```sh
Popen(['/bin/sh', '-c', args[0], args[1], ...])
```
The shell's `-c` option only interprets `args[0]` as the command string while all other arguments are again interpreted by `/bin/sh`. In particular, this means that the string `args[0]` must be fully shell-escaped, so to me this doesn't seem to add any value compared to just calling `Popen()` with the string and `shell=True`.

In short, I believe the convention »pass a string of shell commands or a sequence of command name and arguments« provides the most useful functionality of `Popen()` to Conan scripts.

Reply:
Also note that the Python [documentation](https://docs.python.org/3/library/subprocess.html#popen-constructor) states:
> If *shell* is `True`, it is recommended to pass *args* as a string rather than as a sequence.